### DISCUSSION TITLE
Is Large Page support on Linux only Transparent Huge Pages?

### DISCUSSION THREAD

**Discussion by RogerThiede** [official-stockfish/Stockfish#3887] [Q&A] [answered]
I wanted to benchmark the difference (if any) between Stockfish using transparent huge pages on Linux versus reserved huge pages on Linux when provided 0.5 TB RAM or larger.

README.md:
`Support on Linux - Large page support on Linux is obtained by the Linux kernel transparent huge pages functionality. Typically, transparent huge pages are already enabled, and no configuration is needed.`

I was able to confirm through `/proc/meminfo`  that AnonHugePages ([Transparent Huge Pages](https://developers.redhat.com/blog/2014/03/10/examining-huge-pages-or-transparent-huge-pages-performance)) were being utilized by Stockfish when I `setoption name Hash value 500000`.  However, transparent huge pages feature only supports 2 MB pages, and I am curious if larger than 2MB page sizes provide any added benefit when using 0.5 TB or 1 TB RAM.  Huge Pages  however can be customized to various sizes as opposed to Transparent Huge Pages which are 2MB.

Following instructions from [Configuring Huge Pages](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/monitoring_and_managing_system_status_and_performance/configuring-huge-pages_monitoring-and-managing-system-status-and-performance), I am able to disable transparent huge pages, and reserve many huge pages of 1GB size during boot, but Stockfish never appears to consume them or the OS never seems to allocate them to Stockfish.  Of course 1GB huge pages may be unrealistic, but I wanted to benchmark various sizes to discover any sweet spots.  Stockfish terminates with `Killed` after `setoption name Hash value 500000`.

Is Large Page support on Linux only Transparent Huge Pages, and limited to 2MB huge pages?

**Comment by vondele** [comment, marked as answer]
yes, I think it is, but by way of an experiment might be not too difficult to change, the code is simple 
https://github.com/official-stockfish/Stockfish/blob/44b1ba89a95f394f3e180eb508f2d7798417c86e/src/misc.cpp#L447-L462
